#!/usr/bin/env python 
PACKAGE = "ur5e_test" 

from dynamic_reconfigure.parameter_generator_catkin import * 

gen = ParameterGenerator() 

gen.add("left_rotation", bool_t,0,"left_rotation",False)
gen.add("right_rotation", bool_t,0,"right_rotation",False)
gen.add("ptc", bool_t,0,"ptc",False)
gen.add("torque", bool_t,0,"torque",False)
gen.add("tongue", bool_t,0,"tongue",False)
gen.add("left_force",double_t,0,"left_force",1,0,5) 
gen.add("right_force",double_t,0,"right_force",1,0,5) 
gen.add("left_MX",double_t,0,"left_MX",0.4,0,1.5)
gen.add("left_MY",double_t,0,"left_MY",0.4,0,1.5)  
gen.add("right_MX",double_t,0,"right_MX",0.4,0,1.5)
gen.add("right_MY",double_t,0,"right_MY",0.4,0,1.5)
gen.add("right_angle_My", double_t, 0, "right_angle_My", 0, -60, 60)
gen.add("right_angle_Mx", double_t, 0, "right_angle_Mx", 0, -30, 30)
gen.add("right_angle_Mz", double_t, 0, "right_angle_Mz", 0, -30, 30)

left_record_command = gen.enum([gen.const("l_start", int_t, 1, "force data record starting..."),
                         gen.const("l_stop", int_t, 0, "force data record stopping..."),
                         gen.const("l_clear", int_t, 2, "force data clearing...")], "record command")
gen.add("left_force_recorder", int_t, 0, "record command", 0, 0, 2, edit_method = left_record_command);

right_record_command = gen.enum([gen.const("r_start", int_t, 1, "force data record starting..."),
                         gen.const("r_stop", int_t, 0, "force data record stopping..."),
                         gen.const("r_clear", int_t, 2, "force data clearing...")], "record command")
gen.add("right_force_recorder", int_t, 0, "record command", 0, 0, 2, edit_method = right_record_command);

exit(gen.generate(PACKAGE,"ur5e_test","reference"))
